### PR TITLE
Update to new kvdb/trie/memdb versions

### PR DESCRIPTION
**Pull request by NikVolf**
- No more `ElasticArray`
- Columns are now required, and no "default column" notion exists

### CODE REVIEW

**Comment by bkchr** [comment]
Looks good, but does not compile. What did change in the new trie version?

**Comment by NikVolf** [comment]
> Looks good, but does not compile. What did change in the new trie version?

Yes, doing it already, looks like the version currently published is not compilable to wasm/no-std

**Comment by bkchr** [comment]
I also just want to make sure that we not again break the storage root, because of a trie update ^^

**Comment by NikVolf** [comment]
> What did change in the new trie version?

Haven't seen any logic changes, there were just some api changes (no ElasticArray array now, columns are now `u32`, not `Option`). I'm also currently adding the memory footprint calculations of database structures (#830)

**Comment by bkchr** [comment]
(Please use merge commits)

File: Cargo.lock [hunk 1]
```diff
@@ -8297,7 +8339,8 @@ dependencies = [
 "checksum kernel32-sys 0.2.2 (registry+https://github.com/rust-lang/crates.io-index)" = "7507624b29483431c0ba2d82aece8ca6cdba9382bff4ddd0f7490560c056098d"
 "checksum kv-log-macro 1.0.4 (registry+https://github.com/rust-lang/crates.io-index)" = "8c54d9f465d530a752e6ebdc217e081a7a614b48cb200f6f0aee21ba6bc9aabb"
 "checksum kvdb 0.1.1 (registry+https://github.com/rust-lang/crates.io-index)" = "c1b2f251f01a7224426abdb2563707d856f7de995d821744fd8fa8e2874f69e3"
```
Comment (niklasad1) [inline Cargo.lock]: I think you have missed updating `kvdb-rocksdb` to `v0.3` but shouldn't be related to the errors related `trie-db` I guess

**Comment by NikVolf** [comment]
@bkchr Tested with Kusama, just in case.

**Pull request merged**